1212 ge flow editing

RyanMcilnay opened the pull request:
Closes #1212

RyanMcilnay commented:
@philip-alldredge Fixed issues and did some cleanup, ready for review now.

philip-alldredge commented on ge/org.osate.ge/src/org/osate/ge/internal/ui/editor/FlowContributionItemUtil.java:
```diff
@@ -0,0 +1,203 @@
+import org.osate.ge.internal.ui.tools.FlowDialogUtil;
+import org.osate.ge.internal.util.AadlHelper;
+
+public class FlowContributionItemUtil {
```
Make private. Add private empty constructor if this class contains only static methods.

philip-alldredge commented on ge/org.osate.ge/src/org/osate/ge/internal/ui/editor/DeleteFlowContributionItem.java:
```diff
@@ -0,0 +1,163 @@
+				}
+			}
+
+			private Optional<FlowImplementation> getFlowImplementation(final List<FlowImplementation> flowImpls) {
```
During testing, all flow implementations had the name "<unknown>".  All I did is create 2 flow source implementations and then tried to delete. In practice, the name doesn't make sense in this case. Multiple flow implementations will have the same name. It could be named based like "spec1 implementation #1 (<Component Implementation>)" or something. Not sure what is practical.

RyanMcilnay replied:
Just to make sure I understand, when I tested this shows up as "unknown", which is something I broke while cleaning up and I have now fixed it.

RyanMcilnay replied:
I have added the "#" sign and index as well.

philip-alldredge replied:
I'll retest during the next review.

RyanMcilnay replied:
Okay, I'll go ahead and push my changes.

RyanMcilnay commented on ge/org.osate.ge/src/org/osate/ge/internal/ui/editor/DeleteFlowContributionItem.java:
```diff
@@ -0,0 +1,163 @@
+					// If there are multiple flow implementations for the flow spec, choose desired one.
+					final List<FlowImplementation> flowImpls = componentImpl.getOwnedFlowImplementations().stream()
+							.filter(fi -> fi.getSpecification() == flowSpecification).collect(Collectors.toList());
+					return getFlowImplementation(flowImpls).orElseThrow(() -> new RuntimeException(
```
Good catch!

RyanMcilnay commented on ge/org.osate.ge/src/org/osate/ge/internal/services/ColoringService.java:
```diff
@@ -46,6 +46,8 @@
 		 */
 		void setForeground(DiagramElement de, Color color);
 
+		Color getForeground(DiagramElement de);
```
Did not think of doing it that way, good idea.

philip-alldredge commented:
@RyanMcilnay I pushed a change to `setForeground` which I believe simplifies things. Please take a quick look and let me know if you concur with the change.

RyanMcilnay commented:
@philip-alldredge I noticed End to end flows were not highlighting the first element orange (like it previously did).  This is now fixed.  I also added documentation and screenshots.  As well as removed old flow edit dialog code.
